network/src/commonMain/kotlin/com/wire/kalium/network/api/base/authenticated/self/SelfApi.kt

```diff
@@ -27,7 +29,13 @@ data class ChangeHandleRequest(
     @SerialName("handle") val handle: String
 )
 
-interface SelfApi {
+interface BaseApi {
+    fun getApiNotSupportError(apiName: String, apiVersion: Int) = NetworkResponse.Error(
+        APINotSupported("${this::class.simpleName}: $apiName api is only available on API V$apiVersion")
+    )
+}
```

Comment:
Is this only for `SelfApi`?

It's in `SelfApi.kt`, but its name and behaviour look quite generic.

Reply:
Plan was to also do for other APIs since this function currently duplicated in many places. I'll clean this up.

Reply:
Moved to a separate file.